hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/ListingBasedRollbackStrategy.java

```diff
@@ -57,20 +81,252 @@ public ListingBasedRollbackStrategy(HoodieTable table,
+              if (commitMetadata.getPartitionToWriteStats().containsKey(partitionPath)) {
+                hoodieRollbackRequests.addAll(
+                    getRollbackRequestForAppend(partitionPath, instantToRollback, commitMetadata, table));
+              }
+              break;
+            default:
+              throw new HoodieRollbackException("Unknown listing type, during rollback of " + instantToRollback);
+          }
+        }
+        return hoodieRollbackRequests.stream();
+      }, numPartitions);
+    } catch (Exception e) {
       LOG.error("Generating rollback requests failed for " + instantToRollback.getTimestamp(), e);
       throw new HoodieRollbackException("Generating rollback requests failed for " + instantToRollback.getTimestamp(), e);
     }
   }
+
+  private String getBaseFileExtension(HoodieTableMetaClient metaClient) {
+    return metaClient.getTableConfig().getBaseFileFormat().getFileExtension();
+  }
+
+  @NotNull
+  private HoodieRollbackRequest deleteDataAndLogFilesRollbackRequest(HoodieInstant instantToRollback,
+                                                                     HoodieTableMetaClient metaClient,
+                                                                     String partitionPath)
+      throws IOException {
+    final FileStatus[] filesToDeletedStatus = getBaseAndLogFilesToBeDeleted(metaClient, instantToRollback, partitionPath);
+    return getHoodieRollbackRequest(partitionPath, filesToDeletedStatus);
+  }
+
+  @NotNull
+  private HoodieRollbackRequest deleteDataFilesOnlyRollbackRequest(HoodieInstant instantToRollback,
+                                                                   String basefileExtension, String partitionPath,
+                                                                   FileSystem fs)
+      throws IOException {
+    final FileStatus[] dataFilesToDeletedStatus =
+        getBaseFilesToBeDeleted(instantToRollback.getTimestamp(), basefileExtension,
+            partitionPath, fs);
+    return getHoodieRollbackRequest(partitionPath, dataFilesToDeletedStatus);
+  }
+
+  @NotNull
+  private HoodieRollbackRequest getHoodieRollbackRequest(String partitionPath, FileStatus[] filesToDeletedStatus) {
+    List<String> filesToBeDeleted = getFilesToBeDeleted(filesToDeletedStatus);
+    return new HoodieRollbackRequest(
+        partitionPath, EMPTY_STRING, EMPTY_STRING, filesToBeDeleted, Collections.emptyMap());
+  }
+
+  @NotNull
+  private List<String> getFilesToBeDeleted(FileStatus[] dataFilesToDeletedStatus) {
+    return Arrays.stream(dataFilesToDeletedStatus).map(fileStatus -> {
+      String dataFileToBeDeleted = fileStatus.getPath().toString();
+      // strip scheme
+      return dataFileToBeDeleted.substring(dataFileToBeDeleted.indexOf(":") + 1);
+    }).collect(Collectors.toList());
+  }
+
+  private FileStatus[] getBaseFilesToBeDeleted(String commit, String basefileExtension, String partitionPath,
```

Comment:
Let's make naming more aliased with an actions we're performing, for ex:

 - `getBaseFilesToBeDeleted` > `listFilesToBeDeleted`
 - `getBaseAndLogFilesToBeDeleted` > `fetchFilesFromCommitMetadata`